### DISCUSSION TITLE
Reload form with success message and hide form?

### DISCUSSION THREAD

**Discussion by Coysh** [solspace/craft-freeform#1468] [Questions & Answers] [answered]
I'm sure this used to be an option in older FreeForm versions - but now I can only reload the form with success message or redirect.
I'm using AJAX, but the option doesn't appear either way.

```
{{ freeform.form("register").render({
useRequiredAttribute: true,
}) }}
```

**Comment by kjmartens** [comment]
In Freeform 5, this changed since we now have the [Template Overrides](https://docs.solspace.com/craft/freeform/v5/templates/formatting/#template-overrides) feature.

This can be done one of two different ways now... 🙂 

### At form level

``` twig
{{ form.renderTag({
    fields: {
        ":required": {
            attributes: {
                label: { "+class": "freeform-required" },
                "required": true,
            },
        },
    },
}) }}
```

_OR_

### At field level

``` twig
{{ field.render({
    attributes: {
        label: {
            id: "label-for-" ~ field.handle,
            class: [
                "form-control-label",
                field.required ? "required",
            ],
        },
    },
}) }}
```

**Comment by Coysh** [comment]
@kjmartens thanks, but still not sure how my request would be done as I'm using an AJAX form and want to hide the entire form (except the success message). I'm using the Freeform Tailwind template

**Reply by kjmartens** [reply]
Oh, I am sorry... I somehow focused on the `useRequiredAttribute: true` part of the code in your question... I'll test this on my end and see if I get the same issue.

**Reply by Coysh** [reply]
@kjmartens Thanks!

**Comment by kjmartens** [comment, marked as answer]
Are you saying you're not seeing this option?

<img width="979" alt="Screenshot 2024-08-12 at 5 02 25 PM" src="https://github.com/user-attachments/assets/24cfb6a6-6cfd-49d7-ba60-f8af4cb05387">

---

If so, it's likely because you don't have the "Success Template" directory path setting in the Template Manager area set (or is invalid). Can you verify that you have this configured? 🙂 

<img width="1671" alt="Screenshot 2024-08-12 at 5 04 41 PM" src="https://github.com/user-attachments/assets/c674afb3-625f-4e42-b512-08a894cf1922">

**Reply by Coysh** [reply]
Correct, this is what I'm seeing:
<img width="360" alt="CleanShot 2024-08-12 at 23 36 36@2x" src="https://github.com/user-attachments/assets/dde49129-82cc-4aa6-8c43-0a723992d3cf">

Not sure if it helps, but this is my form row:
```
INSERT INTO `freeform_forms` (`id`, `type`, `name`, `handle`, `spamBlockCount`, `metadata`, `order`, `createdByUserId`, `updatedByUserId`, `dateArchived`, `dateCreated`, `dateUpdated`, `uid`)
VALUES
	(1, 'Solspace\\Freeform\\Form\\Types\\Regular', 'New listing', 'newListing', 0, '{\"behavior\":{\"ajax\":false,\"successBehavior\":\"reload\",\"duplicateCheck\":\"no_limit\",\"showProcessingSpinner\":true,\"successTemplate\":null,\"stopSubmissionsAfter\":null,\"showProcessingText\":false,\"returnUrl\":\"\",\"processingText\":\"Processing...\",\"successMessage\":\"Form has been submitted successfully!\",\"errorMessage\":\"Sorry, there was an error submitting the form. Please try again.\"},\"general\":{\"name\":\"New listing\",\"storeData\":true,\"handle\":\"newListing\",\"defaultStatus\":2,\"type\":\"Solspace\\\\Freeform\\\\Form\\\\Types\\\\Regular\",\"collectIpAddresses\":true,\"sites\":[\"1\"],\"submissionTitle\":\"{{ theTitle }} - {{ dateCreated|date(\\\"Y-m-d H:i:s\\\") }}\",\"allowUsersToOptIn\":false,\"formattingTemplate\":\"ealistings\\/index.twig\",\"optInCheckbox\":null,\"description\":\"\",\"color\":\"#55B97F\",\"attributes\":{\"form\":{},\"row\":{},\"success\":{},\"errors\":{}}}}', NULL, 1, 1, NULL, '2024-06-26 06:54:57', '2024-08-12 17:15:02', 'a0382bdc-fa15-4218-9f27-d94f7f014810');
```

And yes, have the directory path:
<img width="356" alt="CleanShot 2024-08-12 at 23 39 15@2x" src="https://github.com/user-attachments/assets/dc845baf-72ee-439c-99fd-c502d80c3d70">

**Reply by kjmartens** [reply]
Does that path exist and/or is it configured correctly? 🙂 

A quick way to check would be to add a sample template to it using the _Add Sample_ button. My hunch is that it's not configured correctly or you have no templates in that directory because the list of that directory, while cropped in the screenshot, does not seem to extend past `_forms/success`, implying that there are no templates in that directory.

**Reply by Coysh** [reply]
@kjmartens - yep that was it. Thanks so much
